File: TestForBrights.Web/Models/UrlInfoData.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace TestForBrights.Web.Models
{
    public class UrlInfoData
    {
        [Key]
        public int Id { get; set; }
        public string Url { get; set; }
        public DateTime RequestDate { get; set; }
        public string Title { get; set; }
        public string Status { get; set; }
    }
}

File: TestForBrights.Web/ClientApp/components/urlinfo/UrlInfoData.ts
export class UrlInfoData {
    isLoading: boolean = true;
    id: number = 0;
    url: string;
    requestDate: Date | null;
    title: string;
    status: string;

    public constructor(url: string, lastRequestDate: Date | null, lastTitle: string, lastStatus: string) {
        this.url = url;
        this.requestDate = lastRequestDate;
        this.title = lastTitle;
        this.status = lastStatus;
    }

    static getEmptyUrlInfoData(): UrlInfoData {
        return new UrlInfoData('', null, '', '');
    }

    static copyConstructor(source: UrlInfoData, target: UrlInfoData): void {
        target.id = source.id;
        target.url = source.url;
        target.requestDate = source.requestDate;
        target.title = source.title;
        target.status = source.status;
        target.isLoading = source.isLoading;
    }

    getLastRequestDate(): string {
        return this.requestDate ? this.requestDate.toLocaleString('ru-RU') : '';
    }
}
